From 19113e8b2eef8bd7848266878de342b928b40caa Mon Sep 17 00:00:00 2001
From: Nathan Hakkakzadeh <nhakkakzadeh@mozilla.com>
Date: Thu, 2 Jun 2016 13:53:10 -0700
Subject: [PATCH 06/12] Bug 1100925 - For whatever reason MinGW Python cannot
 import functools which is needed in the build process. r=gps

By placing the functools module in the REQUIRED_MODULES list, the environment will not throw an ImportError when the build systemimports functools.

MozReview-Commit-ID: 8DjzXqxvxXd

--HG--
extra : rebase_source : 47ceae1e8b852e0b8bf61f9b34cc953db42e393e

(cherry picked from commit 7394210aad55ad8b863205c1a1862c34783c4080)
Signed-off-by: Jacek Caban <jacek@codeweavers.com>
---
 python/virtualenv/virtualenv.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/python/virtualenv/virtualenv.py b/python/virtualenv/virtualenv.py
index a1854ab36465..442c6e3a5b2e 100755
--- a/python/virtualenv/virtualenv.py
+++ b/python/virtualenv/virtualenv.py
@@ -131,6 +131,8 @@ if majver == 2:
         REQUIRED_MODULES.extend(['warnings', 'linecache', '_abcoll', 'abc'])
     if minver >= 7:
         REQUIRED_MODULES.extend(['_weakrefset'])
+    if is_msys2:
+        REQUIRED_MODULES.extend(['functools'])
 elif majver == 3:
     # Some extra modules are needed for Python 3, but different ones
     # for different versions.
-- 
2.24.1

